### ISSUE TITLE
question about LAMB implementation.

### ISSUE THREAD

**Issue by fastalgo** [mgrankin/over9000#12]
Hi Author, 

Thanks for your implementation on these different optimizers. This repository is great.

By using the LAMB optimizer, I can achieve a higher accuracy than what you reported.

I only updated 2 lines (change LAMB v3 to LAMB v1).

You can easily reproduce my results in this repository:
https://github.com/fastalgo/over9000_lamb

The log files are also in this repository.

**Comment by fastalgo** [comment]
I'm a little surprised that LAMB did not perform well in your experiments.

The TensorFlow implementation of LAMB achieved a good performance for ResNet-50.
You can easily reproduce my results in this repository:
https://github.com/fastalgo/imagenet_resnet50_lamb

Here are some details that may affect the accuracy:

1. LAMB disabled the L2 normalization and used the weight decay. They are different for LAMB.
https://github.com/fastalgo/imagenet_resnet50_lamb/blob/master/resnet_main.py#L352 (here is the L2)

2. It avoided weight decay for batch normalization layers but included weight decay for bias layers.
https://github.com/fastalgo/imagenet_resnet50_lamb/blob/master/resnet_main.py#L352

3. It also avoided trust ratio for the non weight-decay layers.
https://github.com/fastalgo/imagenet_resnet50_lamb/blob/master/resnet_main.py#L352

4. It is did gradient clipping.
https://github.com/fastalgo/imagenet_resnet50_lamb/blob/master/optimization.py#L46

**Comment by mgrankin** [comment]
Amazing job! I haven't tried to reproduce this myself, but PR is very welcomed.

**Comment by Whu-wxy** [comment]
@fastalgo How does LAMB work for small batch size (8/16)?

**Comment by fastalgo** [comment]
> @fastalgo How does LAMB work for small batch size (8/16)?

In my experiments, it works very well

**Comment by Whu-wxy** [comment]
Great! I can't wait to try it!

**Comment by fastalgo** [comment]
> Great! I can't wait to try it!

You may need to change the learning rate. Thanks!
